### PR TITLE
Precland cleanups

### PR DESCRIPTION
**Pull request by amilcarlucas**
Removes code duplication in SITL
Make SITL behave like other backends
Small (mostly NFC) cleanups on the Precland that make the future sensor lag estimation PR smaller.

### CODE REVIEW

**Comment by amilcarlucas** [comment]
```
Binary Name      Text [B]       Data [B]     BSS (B)       Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  -------------  -----------  ------------  ----------------------------  -------------------------
ardurover        36 (+0.0022%)  0 (0.0000%)  4 (+0.0015%)  36 (+0.0022%)                                    337300
blimp            0 (0.0000%)    0 (0.0000%)  0 (0.0000%)   0 (0.0000%)                                      623072
arducopter       32 (+0.0018%)  0 (0.0000%)  0 (0.0000%)   32 (+0.0018%)                                    19[29](https://github.com/ArduPilot/ardupilot/actions/runs/7914430808/job/21604152757?pr=26238#step:10:30)56
arduplane        0 (0.0000%)    0 (0.0000%)  0 (0.0000%)   0 (0.0000%)                                      201296
ardusub          0 (0.0000%)    0 (0.0000%)  0 (0.0000%)   0 (0.0000%)                                      400608
antennatracker   0 (0.0000%)    0 (0.0000%)  0 (0.0000%)   0 (0.0000%)                                      645948
arducopter-heli  40 (+0.0023%)  0 (0.0000%)  0 (0.0000%)   40 (+0.0022%)                                    186780
```

File: libraries/AC_PrecLand/AC_PrecLand.cpp [hunk 1]
```diff
@@ -223,7 +223,7 @@ void AC_PrecLand::init(uint16_t update_rate_hz)
 
     // create inertial history buffer
     // constrain lag parameter to be within bounds
-    _lag.set(constrain_float(_lag, 0.02f, 0.25f));
+    _lag.set(constrain_float(_lag, 0.02f, 0.25f));  // must match LAG parameter range at line 124
```
Comment (Ryanf55) [inline libraries/AC_PrecLand/AC_PrecLand.cpp:226]: Could you put in a static assert to enforce that instead of a comment?

File: libraries/AC_PrecLand/AC_PrecLand.h [hunk 2]
```diff
@@ -228,6 +228,7 @@ class AC_PrecLand
         uint64_t time_usec;
     };
     ObjectArray<inertial_data_frame_s> *_inertial_history;
+    struct inertial_data_frame_s *_inertial_data_delayed;
```
Comment (rishabsingh3003) [inline libraries/AC_PrecLand/AC_PrecLand.h:231]: I am not sure if I understand the rationale behind this change

File: libraries/AC_PrecLand/AC_PrecLand.cpp (hunk 1 above)
Comment (rishabsingh3003) [reply libraries/AC_PrecLand/AC_PrecLand.cpp:226]: The constraint is just so that the inertial buffer doesn't become too big. We could just update the parameter description to be more verbose.. But probably doesn't hurt either to enforce

Comment (rishabsingh3003) [reply libraries/AC_PrecLand/AC_PrecLand.cpp:226]: With some people running CV based prec land which tends to be a bit more laggy, we might want to think about increasing that buffer actually (in another PR).

**Review by rishabsingh3003 — COMMENTED** [review]
Most things look good and seem like its just NFC cleanup. I had one minor comment.
Has this had some testing just to be sure?

File: libraries/AC_PrecLand/AC_PrecLand.h (hunk 2 above)
Comment (amilcarlucas) [reply libraries/AC_PrecLand/AC_PrecLand.h:231]: The next commit requires this change. Doing it here and now makes the next commit a lot easier to understand

**Comment by amilcarlucas** [comment]
This has been tested in CI

File: libraries/AC_PrecLand/AC_PrecLand_SITL.h [hunk 3]
```diff
@@ -33,7 +33,7 @@ class AC_PrecLand_SITL : public AC_PrecLand_Backend
     uint32_t los_meas_time_ms() override { return _los_meas_time_ms; }
 
     // return true if there is a valid los measurement available
-    bool have_los_meas() override;
+    bool have_los_meas() override { return _have_los_meas; }
```
Comment (khancyr) [inline libraries/AC_PrecLand/AC_PrecLand_SITL.h:36]: what is the reason of this change of logic ?

**Comment by tridge** [comment]
needs approval from @rishabsingh3003

**Comment by amilcarlucas** [comment]
I have three other PRs that build on top of this one.

**Comment by amilcarlucas** [comment]
@rishabsingh3003 ping

File: libraries/AC_PrecLand/AC_PrecLand_SITL.h (hunk 3 above)
Comment (khancyr) [reply libraries/AC_PrecLand/AC_PrecLand_SITL.h:36]: Hum
the change is good anyway was L39 is doing the calculation and we are still checking for update in less that 1s.

**Review by khancyr — COMMENTED** [review]
ok for me. 

There is slight behavior check but it is a good addition (L618).
_inertial_data_delayed was move to the the class variable in the promise of a new usage. But I won't bother about this.
rest is code style.

**Review by rishabsingh3003 — APPROVED** [review]
I just ran this locally to confirm there is no change
Looks good to me!
Thanks

**Comment by tridge** [comment]
@amilcarlucas I'd like to delay this till after https://github.com/ArduPilot/ardupilot/pull/26380 is merged as I'd like that in 4.5, and this would make backporting to 4.5.x much harder

**Comment by amilcarlucas** [comment]
@tridge it rebased cleanly
